main.py

```diff
@@ -0,0 +1,105 @@
+                    stats.append([answer._numeric._int_value, answer._owner._value])
+
+                stats.sort(reverse = True)
+
+                print("\nTop 3 countries with most Vaccinations:")
+                for i in range(3):
+                    print(stats[i][1], " : ", stats[i][0])
+
+def vaccineByAge(l, r):
+    
+    with TypeDB.core_client("localhost:1729") as client:
+        with client.session("vaccination_schema", SessionType.DATA) as session:
+            with session.transaction(TransactionType.READ) as read_transaction:
+
+                answer_iterator = read_transaction.query().match_group_aggregate('match $p isa person, has age >= ' + str(l) + ', has age <= ' + str(r) + '; $m isa manufacturer, has vaccine_name $v; $vcc (person: $p, manufacturer: $m) isa vaccinator; get $p, $v; group $v; count;')
+
+                print("\nVaccines given to age group", l, '-', r, "years:")
+
+                for answer in answer_iterator:
+                    print(answer._owner._value, " : ", answer._numeric._int_value)
+
+def vaccineByCountry(country):
+
+    with TypeDB.core_client("localhost:1729") as client:
+        with client.session("vaccination_schema", SessionType.DATA) as session:
+            with session.transaction(TransactionType.READ) as read_transaction:
+
+                country_iterator = read_transaction.query().match('match $p isa person, has country $c; get $c;')
+
+                countries = []
+                for country_pres in country_iterator:
+                    name = country_pres.get("c")
+                    countries.append(name.get_value())
+
+                if country not in countries:
+                    print("Country not found")
+                    return
+
+                answer_iterator = read_transaction.query().match_group_aggregate('match $p isa person, has country = "' + country + '"; $m isa manufacturer, has vaccine_name $v; $vcc (person: $p, manufacturer: $m) isa vaccinator; get $p, $v; group $v; count;')
+
+                print("\nVaccines provided in", country, ":")
+                for answer in answer_iterator:
+                    print(answer._owner._value, " : ", answer._numeric._int_value)
+
+def avgAgeByCountry():
+
+    with TypeDB.core_client("localhost:1729") as client:
+        with client.session("vaccination_schema", SessionType.DATA) as session:
+            with session.transaction(TransactionType.READ) as read_transaction:
+
+                answer_iterator = read_transaction.query().match_group_aggregate('match $p isa person, has country $c, has age $a; get $c, $a; group $c; mean $a;')
+
+                print("\nAverage age of people vaccinated in each country:\n")
+
+                for answer in answer_iterator:
+                    print(answer._owner._value, " : ", round(answer._numeric._float_value, 1), "years")
+    
+
+while (True):
+
+    print("\nMENU")
+    print("1. Display countries with most vaccinations")

```

Comment:
This is a slightly inaccurate description - since this only displays the top 3, it's worth mentioning it explicitly.